native/src/widget/text_input.rs

```diff
@@ -193,8 +209,38 @@ where
             .max_width(self.max_width)
             .height(Length::Units(text_size));
 
+        let text_visual_size = renderer.measure_value(
+            &self.value.to_string(),
+            text_size,
+            self.font,
+        );
+
         let mut text = layout::Node::new(limits.resolve(Size::ZERO));
-        text.move_to(Point::new(padding, padding));
+
+        let x = {
+            let width = limits.max().width;
+            let center = (width / 2.0) - (text_visual_size / 2.0);
+            let right = width - text_visual_size;
```

Comment:
I think if you added the `padding` to `center` and `right` here that would fix the padding issue. However, it seems like text alignment should be handled the same way as it is in the `Text` widget where the actual alignment is deferred to the renderer / whatever text layout lib the renderer is using.

Reply:
Oh!! Got it. But shouldn't padding be applied after the position is figured out. Not sure!! Anyways Thanks I think your suggestion will fix the issue, will try it out tonight! 👍